File: example/basic/5-nested-modules/baz/src/Baz.scala
package baz
import scalatags.Text.all._
import mainargs.{main, ParserForMethods, arg}

object Baz {
  @main
  def main(@arg(name = "bar-text") barText: String,
           @arg(name = "qux-text") quxText: String,
           @arg(name = "baz-text") bazText: String): Unit = {
    foo.qux.Qux.main(barText, quxText)

    val value = p(bazText)
    println("Baz.value: " + value)
  }

  def main(args: Array[String]): Unit = ParserForMethods(this).runOrExit(args)
}

File: example/javalib/builds/1-nested-modules/baz/src/baz/Baz.java
package baz;

import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;
import net.sourceforge.argparse4j.ArgumentParsers;
import net.sourceforge.argparse4j.inf.ArgumentParser;
import net.sourceforge.argparse4j.inf.Namespace;
import foo.Foo;

public class Baz {

  public static void main(String barText, String quxText, String fooText, String bazText) {
    Foo.main(barText, quxText, fooText);
    Context context = new Context();
    context.setVariable("text", bazText);
    String value = new TemplateEngine().process("<p th:text=\"${text}\"></p>", context);
    System.out.println("Baz.value: " + value);
  }

  public static void main(String[] args) throws Exception {
    ArgumentParser parser = ArgumentParsers.newFor("Baz").build();
    parser.addArgument("--bar-text").required(true);
    parser.addArgument("--qux-text").required(true);
    parser.addArgument("--foo-text").required(true);
    parser.addArgument("--baz-text").required(true);

    Namespace res = parser.parseArgs(args);
    main(res.getString("bar_text"), res.getString("qux_text"), res.getString("foo_text"), res.getString("baz_text"));
  }
}
